File: app/src/main/java/com/example/rick_and_morty_android_networking_example_app/domain/models/Character.kt
package com.example.rick_and_morty_android_networking_example_app.domain.models

import java.net.URL

data class Character(
    val id: Int,
    val name: String,
    val status: Status,
    val species: String,
    val type: String,
    val gender: Gender,
    val origin: Location,
    val location: Location,
    val image: URL?,
    val episode: List<String>,
    val url: URL?,
    val created: String
)

File: Rick & Morty MasterDetail App/Rick & Morty MasterDetail App/Domain/Models/Character.swift
//
//  Character.swift
//  Rick & Morty MasterDetail App
//
//  Created by Miguel on 5/18/23.
//

import Foundation

struct Character: Hashable {
    let id: Int
    let name: String
    let status: String
    let species: String
    let type: String
    let gender: String
    let origin: Location
    let location: Location
    let image: URL?
    let episode: [String]
    let url: URL?
    let created: String
    
    static func == (lhs: Character, rhs: Character) -> Bool {
        
        return lhs.id == rhs.id
    }
    
    func hash(into hasher: inout Hasher) {
        
        hasher.combine(id)
    }
}
